man/man8/mount.zfs.8

```diff
@@ -37,68 +35,45 @@
 .Ar mountpoint
 .
 .Sh DESCRIPTION
+The
 .Nm
-is part of the zfsutils package for Linux. It is a helper program that
-is usually invoked by the
+helper is used by
 .Xr mount 8
-or
+to mount filesystem snapshots and
+.Sy mountpoint= Ns Ar legacy
+ZFS filesystems, as well as
 .Xr zfs 8
-commands to mount a ZFS dataset.
+if the
+.Ev Em $ZFS_MOUNT_HELPER
+environment variable is not set for all others –
+you should invoke either of the above in most cases.
```

Comment:
I bumped on the wording here a little bit.  How about this slightly reworded version of your updated text.

```
     The mount.zfs helper is used by mount(8) to mount filesystem snapshots
     and mountpoint=legacy ZFS filesystems, as well as by zfs(8) when the
     $ZFS_MOUNT_HELPER environment variable is not set.  Users should
     should invoke either mount(8) or zfs(8) in most cases.
```

Reply:
Updated with that version